### PR TITLE
DDF-3998 Fix CAS client redirects

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Wraps incoming ServletRequests so they use the correct context paths behind a proxy. 
#### Select relevant component teams: 
@codice/security 
#### Ask 2 committers to review/merge the PR and tag them here.

@brendan-hofmann 
@vinamartin 
### Set up DDF behind a reverse proxy
* Checkout this repo and follow the ReadMe to deploy a reverse proxy: https://github.com/brendan-hofmann/nginx-reverse-proxy/tree/context
* Install DDF with ?dev=true to generate certs for your ip/hostname
* Install the public key of the reverse proxy in the keystore and truststore under the alias localhost by running this command in `etc/keystores`: `openssl s_client -showcerts -connect localhost:YOUR_PORT_80_PORT_ASSIGNED_BY_DOCKER< /dev/null | openssl x509 -outform DER > derp.der; keytool -delete -alias localhost -keystore serverKeystore.jks -storepass changeit; keytool -import -alias localhost -keystore serverKeystore.jks -file derp.der -storepass changeit -noprompt; keytool -delete -alias localhost -keystore serverTruststore.jks -storepass changeit; keytool -import -alias localhost -keystore serverTruststore.jks -file derp.der -storepass changeit -noprompt`
* Configure web context policy. Set `SAML|BASIC` for /admin and `SAML|CAS` for /
* Shutdown DDF
* Configure the external properties in system.properties
* Access DDF at https://localhost:DOCKER_PORT/LOCAL_CONTEXT.

### Verification
Verify you can log into DDF given the reverse proxy setup above. Also, verify, that CAS auth still works with a normal installation.

#### Any background context you want to provide?
DDF's CAS client constructs redirect urls based on the ServletRequests it receives. However, ServletRequests are not aware of the external context that may be used when DDF is deployed behind a proxy, and so CAS will generate incorrect redirect urls.

#### What are the relevant tickets?
[DDF-3998](https://codice.atlassian.net/browse/DDF-3998)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/filter/FilteredRequest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    if (request instanceof FilteredRequest) {
+      FilteredRequest filteredRequest = (FilteredRequest) request;
+      contextPath = filteredRequest.contextPath;
+      requestUri = filteredRequest.requestUri;
+      requestUrl = filteredRequest.requestUrl;
+    } else {
+      URL baseUrl = toBaseUrl(serverName);
+      contextPath =
+          isExternalRequest(baseUrl)
+              ? SystemBaseUrl.EXTERNAL.getRootContext().replaceFirst("/$", "")
```
Comment (ahoffer) [inline platform/security/cas/security-cas-client/src/main/java/org/codice/ddf/security/handler/cas/filter/FilteredRequest.java]: Stripping trailing slashes is so common we should have a library function for it.

**Comment by brendan-hofmann** [comment]
build now

**Comment by SmithJosh** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by SmithJosh** [comment]
build now

**Pull request merged**